Frameworks/Foundation/NSDecimalNumber.mm

```diff
@@ -14,296 +14,334 @@
+- (const char*)objCType {
+    return "d";
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-NSComparisonResult NSDecimalCompare(const NSDecimal* leftOperand, const NSDecimal* rightOperand) {
-    UNIMPLEMENTED();
-    return StubReturn();
++ (id<NSDecimalNumberBehaviors>)defaultBehavior {
+    return [NSDecimalNumberHandler defaultDecimalNumberHandler];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-void NSDecimalCopy(NSDecimal* destination, const NSDecimal* source) {
-    UNIMPLEMENTED();
++ (NSDecimalNumber*)one {
+    static dispatch_once_t __oneOnce;
+    dispatch_once(&__oneOnce, ^{
+        s_one = [[NSDecimalNumber alloc] initWithMantissa:1 exponent:0 isNegative:NO];
+    });
+    return s_one;
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-NSCalculationError NSDecimalDivide(NSDecimal* result,
-                                   const NSDecimal* leftOperand,
-                                   const NSDecimal* rightOperand,
-                                   NSRoundingMode roundingMode) {
-    UNIMPLEMENTED();
-    return StubReturn();
++ (NSDecimalNumber*)zero {
+    static dispatch_once_t __zeroOnce;
+    dispatch_once(&__zeroOnce, ^{
+        s_zero = [[NSDecimalNumber alloc] initWithMantissa:0 exponent:0 isNegative:NO];
+    });
+    return s_zero;
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-BOOL NSDecimalIsNotANumber(const NSDecimal* dcm) {
-    UNIMPLEMENTED();
-    return StubReturn();
++ (NSDecimalNumber*)notANumber {
+    static dispatch_once_t __notANumberOnce;
+    dispatch_once(&__notANumberOnce, ^{
+        s_notANumber = [[NSDecimalNumber alloc] initWithMantissa:0 exponent:0 isNegative:YES];
+    });
+    return s_notANumber;
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-NSCalculationError NSDecimalMultiply(NSDecimal* result,
-                                     const NSDecimal* leftOperand,
-                                     const NSDecimal* rightOperand,
-                                     NSRoundingMode roundingMode) {
-    UNIMPLEMENTED();
-    return StubReturn();
++ (NSDecimalNumber*)maximumDecimalNumber {
+    static dispatch_once_t __maxNumberOnce;
+    dispatch_once(&__maxNumberOnce, ^{
+        s_maxNumber =
+            [[NSDecimalNumber alloc] initWithMantissa:std::numeric_limits<unsigned long long>::max() exponent:_MAX_EXP isNegative:NO];
+    });
+    return s_maxNumber;
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-NSCalculationError NSDecimalMultiplyByPowerOf10(NSDecimal* result, const NSDecimal* number, short power, NSRoundingMode roundingMode) {
-    UNIMPLEMENTED();
-    return StubReturn();
++ (NSDecimalNumber*)minimumDecimalNumber {
+    static dispatch_once_t __minNumberOnce;
+    dispatch_once(&__minNumberOnce, ^{
+        s_minNumber =
+            [[NSDecimalNumber alloc] initWithMantissa:std::numeric_limits<unsigned long long>::max() exponent:_MAX_EXP isNegative:YES];
+    });
```

Comment:
why did you choose dispatch_once as opposed to c++ function local statics? Seems easier (and more efficient) to do the latter to me ...